compiler_gym/datasets/benchmark.py

```diff
@@ -0,0 +1,350 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import re
+from concurrent.futures import as_completed
+from pathlib import Path
+from typing import Callable, Iterable, List, NamedTuple, Optional, Union
+
+from compiler_gym.service.proto import Benchmark as BenchmarkProto
+from compiler_gym.service.proto import File
+from compiler_gym.util import thread_pool
+from compiler_gym.util.decorators import memoized_property
+from compiler_gym.validation_result import ValidationError
+
+# A validation callback is a function that takes a single CompilerEnv instance
+# as its argument and returns an iterable sequence of zero or more
+# ValidationError tuples.
+ValidationCallback = Callable[["CompilerEnv"], Iterable[ValidationError]]  # noqa: F821
+
+
+# Regular expression that matches the full two-part URI prefix of a dataset:
+#     <protocol>://<dataset>
+#
+# E.g. "benchmark://foo-v0".
+DATASET_NAME_RE = re.compile(
+    r"(?P<dataset>(?P<dataset_protocol>[a-zA-z0-9-_]+)://(?P<dataset_name>[a-zA-z0-9-_]+-v(?P<dataset_version>[0-9]+)))"
+)
+
+# Regular expression that matches the full three-part format of a benchmark URI:
+#     <protocol>://<dataset>/<id>
+#
+# E.g. "benchmark://foo-v0/" or "benchmark://foo-v0/program".
+BENCHMARK_URI_RE = re.compile(
```

Comment:
I'd change DATASET_NAME_RE to accept both benchmark://foo-v0 and benchmark://foo-v0/, and BENCHMARK_URI_RE to require all 3 parts, i.e benchmark://foo-v0/program

Reply:
Good idea, done